hack/local-up-cluster.sh

```diff
@@ -1086,8 +1086,15 @@ if [[ "${START_MODE}" != "kubeletonly" ]]; then
   kube::etcd::validate
 fi
 
-if [ "${CONTAINER_RUNTIME}" == "docker" ] && ! kube::util::ensure_docker_daemon_connectivity; then
-  exit 1
+if [ "${CONTAINER_RUNTIME}" == "docker" ]; then
+  if ! kube::util::ensure_docker_daemon_connectivity; then
+    exit 1
+  else
+    # docker doesn't allow to reach exposed hostPorts from the node, however, Kubernetes does
+    # so we append a new rule on top of the docker one
+    # -A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER  <-- docker rule
+    iptables -t nat -A OUTPUT -m addrtype --dst-type LOCAL -j DOCKER
```

Comment:
is this something we should clean up on exit? does this tolerate being run multiple times in a row if this rule already exists?

Reply:
heh, good question.
every time you exec the command appends a new rule, 
```
-A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
-A OUTPUT -m addrtype --dst-type LOCAL -j DOCKER
-A OUTPUT -m addrtype --dst-type LOCAL -j DOCKER
-A OUTPUT -m addrtype --dst-type LOCAL -j DOCKER
-A OUTPUT -m addrtype --dst-type LOCAL -j DOCKER
-A OUTPUT -m addrtype --dst-type LOCAL -j DOCKER
-A OUTPUT -m addrtype --dst-type LOCAL -j DOCKER
```

so we should check first if the rule alredy exist
```sh
if ! iptables -t nat -C OUTPUT -m addrtype --dst-type LOCAL -j DOCKER; then 
  iptables -t nat -A OUTPUT -m addrtype --dst-type LOCAL -j DOCKER
fi
```

to delete just swallow the error, it will fail if the chain doesn't exist and delete it if exist
```
$ iptables -t nat -D OUTPUT -m addrtype --dst-type LOCAL -j DOCKER
$ iptables -t nat -D OUTPUT -m addrtype --dst-type LOCAL -j DOCKER
iptables: Bad rule (does a matching rule exist in that chain?).
```

Reply:
Done! thanks.